torch/_prims_common/__init__.py

```diff
@@ -48,8 +48,8 @@ def __rmul__(self, other: Any) -> typing.Self:
     _IntLikeT = TypeVar("_IntLikeT", bound=_WorksWithInt)
 
 
-ShapeType: TypeAlias = Union[torch.Size, list[int], tuple[int, ...]]
-StrideType: TypeAlias = Union[list[int], tuple[int, ...]]
+ShapeType: TypeAlias = Union[torch.Size, Sequence[int], tuple[int, ...]]
+StrideType: TypeAlias = Union[Sequence[int], tuple[int, ...]]
```

Comment:
Ugh, doesn't this just resolve to sequence int now as tuple is a subclass of it? Does this break other things now? Also want to make sure Tensor is not accidentally a subtype of sequence here

Reply:
Urg, you're right, how silly.

In fact, because `torch.Size` is-a `tuple[int, ...]`  both of them resolve to this:

```
ShapeType: TypeAlias = Sequence[int]
StrideType: TypeAlias = Sequence[int]
```

This is actually not so bad, now is it? 🙂 

The previous change didn't cause any mypy issues, we'll see what this does. :-)

Reply:
True, but this will resolve to sequences that are not a list or tuple which could be too permissive in this case.